lib/src/types/relay.ts

```diff
@@ -64,40 +63,50 @@ export class WakuRelay implements IRelay {
 
     async init(): Promise<void> {
         // TODO error flow
-        this.waku = await Waku.create({ bootstrap: true, relayKeepAlive: 59, pingKeepAlive: 59 });
+        return new Promise((resolve) => {
+            // this.socket = new WebSocket('ws://localhost:9001');
+            this.socket = new WebSocket('ws://159.223.102.226:9001');
 
-        const nodes = await getBootstrapNodes();
-        await Promise.all(nodes.map((addr) => {
-            this.waku.dial(addr)
-        }));
+            this.socket.onopen = () => {
+                if (this.hc.debug) console.log("hashconnect - connected");
+                resolve();
+            };
+        });
 
-        if (this.hc.debug) console.log("hashconnect - bootstrapped nodes", nodes);
-        if (this.hc.debug) console.log("hashconnect - Waiting for peer...");
-        await this.waku.waitForConnectedPeer();
-
-        if (this.hc.debug) console.log("hashconnect - connected");
     }
 
     async subscribe(topic: string): Promise<void> {
-        if (this.hc.debug) console.log("hashconnect - Subscribing to " + topic);
-        this.waku.relay.addObserver(this.processMessage, [topic])
+        if (this.hc.debug) console.log("hashconnect - Subscribing to topic id " + topic);
+        this.socket.send(JSON.stringify({ action: 'sub', topic: topic }));
+
+        this.socket.onmessage = (e) => {
+            this.processMessage(e);
+        };
     }
 
-    addDecryptionKey(privKey: string) {
-        if (this.hc.debug) console.log("hashconnect - Adding decryption key " + privKey);
-        this.waku.addDecryptionKey(Buffer.from(privKey, 'base64'))
+    addDecryptionKey(privKey: string, topic: string) {
+        console.log("hashconnect - Adding decryption key \n PrivKey: " + privKey);
+        if (this.hc.debug) console.log("hashconnect - Adding decryption key \n PrivKey: " + privKey);
+        this.hc.publicKeys[topic] = privKey;
```

Comment:
Same here. Maybe it's intentional and the variable name `publicKeys` just needs changing?